### PR TITLE
General improvements and fixes

### PR DESCRIPTION
**Pull request by juanjux**
Same as yesterday PR with feedback incorporated, created this new branch to avoid some rebasing hell on that PR.

### CODE REVIEW

**Comment by juanjux** [comment]
Looks like the build failed because it can't find some file that the own build auto generates, could somebody try to repeat the build? (I don't seem to have permissions).

**Comment by smola** [comment]
@juanjux done, it seems it was a temporal failure: https://travis-ci.org/bblfsh/documentation/builds/240718867

It seems there are some broken links.

**Comment by smola** [comment]
@juanjux you can verify the site with `make build && docker run -t -v $(pwd)/_book:/site 18fgsa/html-proofer /site`

**Comment by juanjux** [comment]
@smola, I did, but I forgot to `git add user/server-protocol.md` which was on my disk so the `make` test worked on my machine®.

File: architecture.md [hunk 1]
```diff
@@ -16,11 +19,20 @@ graph LR
 
 ## Language Drivers
 
-A language driver is a Docker container that takes file parsing requests and respond with the AST and the [UAST](./uast/specification.md).
+A language driver is a program, containerized with Docker, that takes file parsing
+requests with source code and replies with an [UAST](./uast/specification.md)
+representation of the provided source code or an error message.
 
-Our implementations of language drivers are composed of two pieces: an AST parser, written in the source language and an AST to UAST normalizer written in Go. In the case of AST parsers written in Go, AST parser and UAST normalizer might be combined in a single executable.
+Our implementations of language drivers are composed of two pieces: an AST parser,
```
Comment (abeaumont) [inline architecture.md]: s/an AST parser/a code parser/

File: architecture.md [hunk 2]
```diff
@@ -16,11 +19,20 @@ graph LR
...
+that can be written in any language (usually the source language) and an AST to
+UAST normalizer written in Go. In the case of AST parsers written in Go or other
```
Comment (abeaumont) [inline architecture.md]: s/AST parsers/parsers/

File: architecture.md [hunk 3]
```diff
@@ -16,11 +19,20 @@ graph LR
...
+languages producing linkable object files or shared library, AST parser and UAST
```
Comment (abeaumont) [inline architecture.md]: s/AST parser and UAST normalizer/the parser and normalizer/

File: architecture.md [hunk 4]
```diff
@@ -16,11 +19,20 @@ graph LR
...
+normalizer might be combined by dynamic or static linking without a separate
+runnable component.
 
-The entry point of the container is the Go UAST normalizer, which wraps the language-specific AST parser. This is how the Python driver looks like:
+The entry point of the container is the Go UAST normalizer, which communicates
+internally with the language-specific AST parser. This is how the Python driver
```
Comment (abeaumont) [inline architecture.md]: s/AST//

**Pull request merged**